lib/rules/no-extra-parens.js

```diff
@@ -470,11 +499,7 @@ module.exports = {
              * If `node.superClass` is a LeftHandSideExpression, parentheses are extra.
              * Otherwise, parentheses are needed.
              */
-            const hasExtraParens = precedence(node.superClass) > PRECEDENCE_OF_UPDATE_EXPR
-                ? hasExcessParens(node.superClass)
-                : hasDoubleExcessParens(node.superClass);
-
-            if (hasExtraParens) {
+            if (hasExcessParensWithPrecedence(node.superClass, PRECEDENCE_OF_UPDATE_EXPR)) {
```

Comment:
This changes checking precedence from `>` to `>=`, so the rule would now remove parens and produce parsing error in e.g.:

```js
class A extends (++x) {}
```